WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -31,26 +28,45 @@ class MediaCoordinator: NSObject {
     /// - parameter asset: The asset to add.
     /// - parameter blog: The blog that the asset should be added to.
     ///
-    func addMedia(from asset: ExportableAsset, to blog: Blog) {
-        guard let asset = asset as? PHAsset else {
-            return
-        }
+    @discardableResult
+    func addMedia(from asset: ExportableAsset, to blog: Blog) -> Media {
+        return self.addMedia(from: asset, to: blog.objectID)
+    }
+
+    /// Adds the specified media asset to the specified post. The upload process
+    /// can be observed by adding an observer block using the `addObserver(_:for:)` method.
+    ///
+    /// - parameter asset: The asset to add.
+    /// - parameter post: The post that the asset should be added to.
+    ///
+    @discardableResult
+    func addMedia(from asset: ExportableAsset, to post: AbstractPost) -> Media {
+        return self.addMedia(from: asset, to: post.objectID)
+    }
+
+    @discardableResult
+    private func addMedia(from asset: ExportableAsset, to objectID: NSManagedObjectID) -> Media {        
         mediaProgressCoordinator.track(numberOfItems: 1)
         let service = MediaService(managedObjectContext: backgroundContext)
         let totalProgress = Progress.discreteProgress(totalUnitCount: MediaExportProgressUnits.done)
         var creationProgress: Progress? = nil
         let media = service.createMedia(with: asset,
-                            objectID: blog.objectID,
+                            objectID: objectID,
                             progress: &creationProgress,
-                            thumbnailCallback: nil,
+                            thumbnailCallback: { [weak self]media, url in
+                                guard let strongSelf = self else {
+                                    return
+                                }
+                                strongSelf.thumbnailReady(url: url, for: media)
```

Comment:
Instead of guarding for `strongSelf` here, we can just do `self?.thumbnailReady(url: url, for: media)`